pkg/operators/v1/operatorgroup_types.go

```diff
@@ -45,6 +64,29 @@ type OperatorGroupSpec struct {
 	// Static tells OLM not to update the OperatorGroup's providedAPIs annotation
 	// +optional
 	StaticProvidedAPIs bool `json:"staticProvidedAPIs,omitempty"`
+
+	// UpgradeStrategy defines the upgrade strategy for operators in the namespace.
+	// There are currently two supported upgrade strategies:
+	//
+	// Default: OLM will only allow clusterServiceVersions to move to the replacing
+	// phase from the succeeded phase. This effectively means that OLM will not
+	// allow operators to move to the next version if an installation or upgrade
+	// has failed.
+	//
+	// TechPreviewUnsafeFailForward: OLM will allow clusterServiceVersions to move to the
+	// replacing phase from the succeeded phase or from the failed phase.
+	// Additionally, OLM will generate new installPlans when a subscription references
+	// a failed installPlan and the catalog has been updated with a new upgrade for
+	// the existing set of operators.
+	//
+	// WARNING: The TechPreviewUnsafeFailForward upgrade strategy is unsafe and may result
+	// in unexpected behavior or unrecoverable data loss unless you have deep
+	// understanding of the set of operators being managed in the namespace.
+	//
+	// +kubebuilder:validation:Enum=Default;TechPreviewUnsafeFailForward
+	// +kubebuilder:default=Default
+	// +optional
+	UpgradeStrategy string `json:"upgradeStrategy,omitempty"`
```

Comment:
I think @njhale might've mentioned this but should we have a dedicated string type for the UpgradeStrategy here?

Reply:
Sure.

Reply:
Created a string alias named `UpgradeStrategy`.